### PR TITLE
Feature/enhacement configuration

### PR DESCRIPTION
**Pull request by nsmsanchez**
* Rematch used 
* Basic auth model added
* eslint configuration
* prettier configuration
* vite configuration

### CODE REVIEW

File: frontend/src/state/models/auth.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  reducers: {
+    SET_AUTHENTICATION: (state, token: string | null = null) => {
+      console.log('called');
+      return { ...state, token };
+    },
+  },
+  effects: (dispatch) => ({
+    async loginRequest(payload: { username: string; password: string }) {
+      const { username, password } = payload;
+      // dispatch.auth.setAuthentication(token);
```
Comment (nsmsanchez) [inline frontend/src/state/models/auth.ts:27]:
```suggestion
```

File: frontend/index.html [hunk 2]
```diff
@@ -0,0 +1,13 @@
+<!doctype html>
+<html lang="en">
+  <head>
+    <meta charset="UTF-8" />
+    <link rel="icon" type="image/svg+xml" href="/vite.svg" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <title>Vite + React + TS</title>
```
Comment (nsmsanchez) [inline frontend/index.html:7]: Lets replace this by a ENV variable with vite.

**Review by nsmsanchez — COMMENTED** [review]
By somereason still research linter(eslint) is not taken absolute imports in order to make cleaner imports in code base, at this moment is nothing to make us stop.

**Comment by nsmsanchez** [comment]
It was added jest and its configuration, ready to start integrate API service library

**Review by fabrizzioalco — APPROVED** [review]
LGTM, I will provide a mock endpoint for user signup and login. We can merge this so we can keep on this work!

**Comment by nsmsanchez** [comment]
Keeping this PR cleaner as possible to add in other the api manager configuration enhacement

**Pull request merged**